docs/DeveloperGuide.md

```diff
@@ -1,11 +1,118 @@
 # Developer Guide
 
-## Design & implementation
+## Design
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Architecture
+![Architect class diagram](UML diagrams/OverviewClassDiagram.png)
```

Comment:
The architecture diagram seems too granular, e.g. Email and Command seem to be individual classes rather than an entire component. Also, such apps usually use n-tier architecture so it might be better to make the layered nature more explicit in the architecture diagram.